app/lib/pre_assembly/batch.rb

```diff
@@ -48,30 +47,36 @@ def object_filenames_unique?(dobj)
 
     # Discovers the digital object containers and the stageable items within them.
     # For each container, creates a new PreAssembly::Digitalobject.
-    # @return [Array<PreAssembly::DigitalObject>]
+    # @return [Enumerable<PreAssembly::DigitalObject>]
+    # @yield [PreAssembly::DigitalObject]
     # rubocop:disable Metrics/AbcSize
-    def digital_objects
-      @digital_objects ||= containers_via_manifest.each_with_index.map do |c, i|
-        stageable_items = discover_items_via_crawl(c)
+    def digital_objects(&_block)
+      return enum_for(:digital_objects) { containers_via_manifest.size } unless block_given?
+
+      containers_via_manifest.each_with_index.map do |container, i|
```

Comment:
NBD, but for clarity I might look to replace `each_with_index` with `with_index`.